packages/trpc/server/routers/viewer/eventTypes/update.handler.ts

```diff
@@ -302,40 +303,84 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
       break;
     }
   }
-
-  const connectedLink = await ctx.prisma.hashedLink.findFirst({
+  const connectedLinks = await ctx.prisma.hashedLink.findMany({
     where: {
       eventTypeId: input.id,
     },
     select: {
       id: true,
+      link: true,
+      destroyOnUse: true,
     },
   });
 
-  if (hashedLink) {
-    // check if hashed connection existed. If it did, do nothing. If it didn't, add a new connection
-    if (!connectedLink) {
-      // create a hashed link
-      await ctx.prisma.hashedLink.upsert({
+  const connectedPrivateLink = connectedLinks.find((link) => {
```

Comment:
Updating event type removes all single use links even if it wasn't changed


https://github.com/user-attachments/assets/1a0d5636-594b-4d5f-a8a7-ab30ce7ea1ee

Reply:
Fixed, now! Missed this testing from my side. 🙇